core/app/models/spree/variant.rb

```diff
@@ -100,12 +100,12 @@ class Variant < Spree::Base
       end
     end
 
-    self.whitelisted_ransackable_associations = %w[option_values product prices default_price]
+    self.whitelisted_ransackable_associations = %w[option_values product_name prices default_price]
```

Comment:
@imgarylai we need to keep this as `product` as this is for associations, if we need `product_name` attribute we should add it to `whitelisted_ransackable_attributes`